classes/Course/class.hubCourse.php

```diff
@@ -203,8 +203,8 @@ protected function updateAdditionalFields() {
 
 
 	protected function deleteCourse() {
+        $hist = $this->getHistoryObject();
```

Comment:
I think it's the right way to handle this as you implemented it in your PR. but we have installations which run for years with the old implementation and i have to check if they run as before.